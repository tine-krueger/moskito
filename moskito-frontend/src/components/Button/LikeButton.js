import styled from 'styled-components/macro'
import { AiFillPushpin } from 'react-icons/ai'
import { bookmark } from '../../services/handleBookmarkApi'
import { useState } from 'react'


export default function LikeButton({id}) {
    const [ pinned, setPinned] = useState(false)
    

    return <CampsiteLikeButton pinned={pinned} onClick={handleClick}><AiFillPushpin/></CampsiteLikeButton>

    function handleClick() {
        setPinned(!pinned)
        bookmark(id)
        
    }
}

const CampsiteLikeButton = styled.div`
    position: absolute;
    right: 1em;
    top: 1em;
    height: 3.5em;
    width: 3.5em;
    border-radius: 10px;
    box-shadow: 2px 4px 8px -4px rgba(0, 0, 0, 0.6), 
                -2px -2px 2px 0px rgba(255, 255, 255, 0.5);
    display: flex;
    justify-content: center;
    svg {
        width: 2.1em;
        height: 2.1em;
        stroke: #c97f63;
        stroke-width: 3em;
        fill: ${pops => pops.pinned ? '#c97f63' : '#c2d6d3'};
        align-self: center;
        margin-bottom: 3px;
    }
`