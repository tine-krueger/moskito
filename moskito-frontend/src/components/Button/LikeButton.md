```js
import { AiFillPushpin } from 'react-icons/ai';
const [ isPinned, setIsPinned ] = React.useState(false);

<div style={{width: '100px', position: 'relative', height: '100px'}}>
    <LikeButton isPinned={isPinned} onClick={() => setIsPinned(!isPinned)}>
        <AiFillPushpin/>
    </LikeButton>
</div>
```
