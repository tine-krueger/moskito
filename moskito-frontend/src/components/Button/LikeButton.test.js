import { render } from '@testing-library/react'
import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import LikeButton from './LikeButton'

const campsite = {
    pinned: false,
}

const mockOnClick = jest.fn()

const button = <LikeButton
isPinned={campsite.pinned}
onClick={mockOnClick}
/>

const truthyButton = <LikeButton
isPinned={!campsite.pinned}
onClick={mockOnClick}
/>

describe ('Like Button', () =>{
    it('renders correctly', () => {
        const { container } = render(button)
        expect(container).toMatchSnapshot()
    })
    it('pin turns orange, when pinned equals true', () => {
        const { getByTestId } = render(truthyButton)
        expect(getByTestId('pin')).toHaveStyle('fill: var(--link)')
    })

    it('pin turns transparent, when pinned equals false', () => {
        const { getByTestId } = render(button)
        expect(getByTestId('pin')).toHaveStyle('fill: var(--background)')
    })

    it('calls onClick, when Button in clicked', () => {
        const { getByTestId } = render(button)
        userEvent.click(getByTestId('like-button'))
        expect(mockOnClick).toHaveBeenCalledTimes(1)
      })
})