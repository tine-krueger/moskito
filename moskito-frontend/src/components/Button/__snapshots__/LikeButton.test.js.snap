// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like Button renders correctly 1`] = `
.c0 {
  position: absolute;
  right: 1em;
  top: 1em;
  height: 2.5em;
  width: 2.5em;
  border-radius: 10px;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 svg {
  width: 1.7em;
  height: 1.7em;
  stroke: var(--link);
  stroke-width: 3em;
  fill: var(--background);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 3px;
}

<div>
  <div
    class="c0"
    data-testid="like-button"
  >
    <svg
      data-testid="pin"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3-15.4 12.3-16.6 35.4-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 0 0-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8z"
      />
    </svg>
  </div>
</div>
`;
